ccp/lib/clients/openshift/client.py

```diff
@@ -0,0 +1,246 @@
+        c2 = c2 + "'{{ if .data.token }}{{ .data.token }}{{end}}'"
+        c3 = "head -n 1"
+        c4 = "base64 -d -"
+        command = "{} | {} | {} | {}".format(
+            c1, c2, c3, c4
+        )
+        return self.run_command(command, shell=True)
+
+    @retry(tries=5, delay=3, backoff=3)
+    def get_token_from_mounted_secret(
+            self, secret_mount_path="/tmp/jenkins-secret"
+    ):
+        token_location = secret_mount_path + "/token"
+        if not path.exists(token_location):
+            raise Exception(
+                "Could not find token at {}".format(
+                    secret_mount_path
+                )
+            )
+        with open(token_location, "r") as f:
+            return "".join(f.readlines())
+
+    def login(self, server=None, token=None, username=None, password=None):
+        """
+        Logs into an openshift cluster with token or username and password.
+        :param server: Default None: The openshift server to log into. If none
+        is provided then we assume oc is already aware which server it needs
+        to log into
+        :type server str
+        :param token: Default None: The token to login with. Overrides username
+        and password as preferred login method
+        :type token str
+        :param username: Default none : The username to login with. Use only if
+        token is not used.
+        :type username str
+        :param password: The password to login with. Use only if token is not
+        provided. Not recommended
+        :type password str
+        :return: The output ot the command
+        :rtype str
+        :raises Exception
+        """
+        command = str.format(
+            "{base_command} login{token_param}{user_param}{server_param}",
+            base_command=self.base_command,
+            token_param="" if not token else " --token={}".format(
+                token
+            ),
+            user_param="" if token else " -u {} -p {}".format(
+                username,
+                password
+            ),
+            server_param="" if not server else str(server)
+        )
+        return self.run_command(cmd=command)
+
+    @retry(tries=10, delay=3, backoff=2)
+    def process_template(
+            self, template_path, params, namespace, apply_template=True
+    ):
+        """
+        Processes a specified template, and applies to to a namespace
+        :param template_path: The path to the template file to process
+        :param params: The key value pair of parameters with which the template
+        is to be processed
+        :param namespace: The namespace in which the template is to be applied
+        :param apply_template: Default True: Apply the template
+        :type template_path str
+        :type params dict
+        :type namespace str
+        :type apply_template bool
+        :returns output of command execution
+        :rtype str
+        :raises Exception
+        """
```

Comment:
Should we add a check here to verify if a file exists at given `template_path` or not?

Reply:
Added